x/tokenfactory/keeper/admins_test.go

```diff
@@ -185,82 +185,72 @@ func (suite *KeeperTestSuite) TestBurnDenom() {
+				return types.NewMsgChangeAdmin(suite.TestAccs[1].String(), denom, suite.TestAccs[2].String())
 			},
+			expectedAdminIndex: 0,
 		},
 		{
 			desc: "success change admin",
-			setup: func() {
-				// Test Change Admin
-				_, err := suite.msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), types.NewMsgChangeAdmin(suite.TestAccs[0].String(), denom, suite.TestAccs[1].String()))
-				queryRes, err := suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
-					Denom: denom,
-				})
-				suite.Require().NoError(err)
-				suite.Require().Equal(suite.TestAccs[1].String(), queryRes.AuthorityMetadata.Admin)
-
-				// New admin can mint.
-				_, err = suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[1].String(), sdk.NewInt64Coin(denom, 5)))
-
-				addr1bal += 5
-				suite.Require().NoError(err)
-				suite.Require().Equal(suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], denom).Amount.Int64(), addr0bal)
+			msgChangeAdmin: func(denom string) *types.MsgChangeAdmin {
+				return types.NewMsgChangeAdmin(suite.TestAccs[0].String(), denom, suite.TestAccs[1].String())
+			},
+			expectedAdminIndex: 1,
+			msgMint: func(denom string) *types.MsgMint {
+				return types.NewMsgMint(suite.TestAccs[1].String(), sdk.NewInt64Coin(denom, 5))
 			},
 		},
 	} {
 		suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
 			// setup test
 			suite.SetupTest()
-			addr0bal = 0
-			addr1bal = 0
 
 			// Create a denom and mint
 			res, err := suite.msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom(suite.TestAccs[0].String(), "bitcoin"))
 			suite.Require().NoError(err)
-			denom = res.GetNewTokenDenom()
 
-			_, err = suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(denom, 10)))
-			addr0bal += 10
+			testDenom := res.GetNewTokenDenom()
+
+			_, err = suite.msgServer.Mint(sdk.WrapSDKContext(suite.Ctx), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(testDenom, 10)))
+			suite.Require().NoError(err)
+
+			_, err = suite.msgServer.ChangeAdmin(sdk.WrapSDKContext(suite.Ctx), tc.msgChangeAdmin(testDenom))
+			suite.Require().NoError(err)
+
+			queryRes, err := suite.queryClient.DenomAuthorityMetadata(suite.Ctx.Context(), &types.QueryDenomAuthorityMetadataRequest{
+				Denom: testDenom,
+			})
 			suite.Require().NoError(err)
-			suite.Require().True(suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], denom).Amount.Int64() == addr0bal, suite.App.BankKeeper.GetBalance(suite.Ctx, suite.TestAccs[0], denom))
 
-			tc.setup()
+			// expectedAdminIndex with negative value is assumed as admin with value of ""
+			if tc.expectedAdminIndex < 0 {
+				suite.Require().Equal("", queryRes.AuthorityMetadata.Admin)
+			} else {
+				suite.Require().Equal(suite.TestAccs[tc.expectedAdminIndex].String(), queryRes.AuthorityMetadata.Admin)
+			}
+
+			if tc.msgMint != nil {
```

Comment:
Let's add a comment explaining why we are also testing mint when the test spec is focused on the change admin message. I think the goal here is to make sure that we can mint after changing admin?

Reply:
Yeah, basically this is our "test" for the authority actually being correct

Reply:
added a comment here!